src/goImport.ts

```diff
@@ -9,31 +9,95 @@ import vscode = require('vscode');
 
 export function listPackages(excludeImportedPkgs: boolean = false): Thenable<string[]> {
 	let importsPromise = excludeImportedPkgs && vscode.window.activeTextEditor ? getImports(vscode.window.activeTextEditor.document.fileName) : Promise.resolve([]);
-	let pkgsPromise = new Promise<string[]>((resolve, reject) => {
+	let vendorSupportPromise = isVendorSupported();
+	let goPkgsPromise = new Promise<string[]>((resolve, reject) => {
 		cp.execFile(getBinPath('gopkgs'), [], (err, stdout, stderr) => {
 			if (err && (<any>err).code === 'ENOENT') {
 				promptForMissingTool('gopkgs');
 				return reject();
 			}
 			let lines = stdout.toString().split('\n');
-			let sortedlines = lines.sort().slice(1); // Drop the empty entry from the final '\n'
-			return resolve(sortedlines);
+			return resolve(lines);
 		});
 	});
 
-	return Promise.all<string[]>([importsPromise, pkgsPromise]).then(values => {
-		let imports = values[0];
-		let pkgs = values[1];
-		if (imports.length === 0) {
-			return pkgs;
-		}
-		return pkgs.filter(element => {
-			return imports.indexOf(element) === -1;
+	return vendorSupportPromise.then((vendorSupport: boolean) => {
+		return Promise.all<string[]>([goPkgsPromise, importsPromise]).then(values => {
+			let pkgs = values[0];
+			let importedPkgs = values [1];
+
+			if (!vendorSupport || !vscode.window.activeTextEditor) {
+				if (importedPkgs.length > 0) {
+					pkgs = pkgs.filter(element => {
+						return importedPkgs.indexOf(element) === -1;
+					});
+				}
+				return pkgs.sort().slice(1);
+			}
+
+			let currentFileDirPath = path.dirname(vscode.window.activeTextEditor.document.fileName);
+			let workspaces: string[] = process.env['GOPATH'].split(path.delimiter);
+			let currentWorkspace = path.join(workspaces[0], 'src');
+
+			// Workaround for issue in https://github.com/Microsoft/vscode/issues/9448#issuecomment-244804026
+			if (process.platform === 'win32') {
+				currentFileDirPath = currentFileDirPath.substr(0, 1).toUpperCase() + currentFileDirPath.substr(1);
+			}
+
+			// In case of multiple workspaces, find current workspace by checking if current file is
+			// under any of the workspaces in $GOPATH
+			for (let i = 1; i < workspaces.length; i++) {
+				let possibleCurrentWorkspace = path.join(workspaces[i], 'src');
+				if (currentFileDirPath.startsWith(possibleCurrentWorkspace)) {
+					// In case of nested workspaces, (example: both /Users/me and /Users/me/src/a/b/c are in $GOPATH)
+					// both parent & child workspace in the nested workspaces pair can make it inside the above if block
+					// Therefore, the below check will take longer (more specific to current file) of the two
+					if (possibleCurrentWorkspace.length > currentWorkspace.length) {
+						currentWorkspace = possibleCurrentWorkspace;
+					}
+				}
+			}
+
+			let updatedPackages: string[] = [];
+			pkgs.forEach(pkg => {
+				if (!pkg || importedPkgs.indexOf(pkg) > -1) {
+					return;
+				}
+
+				let vendorIndex = pkg.indexOf('/vendor/');
```

Comment:
Below is from [Understanding and using the vendor folder](https://blog.gopheracademy.com/advent-2015/vendor-folder/)
>  However, if a package or a parent folder of a package contains folder named vendor it will be searched for dependencies using the vendor folder as an import path root. 

And while trying different combinations, I could confirm that by "parent", they mean "ancestor"
Since that is the exact check I make which is making sure that the "vendor" folder's ancestors are current package's ancestors too, we should be good.

Go doesn't stop you from creating a package by the name "vendor", so you could totally do a "go get github.com/foo/vendor". Any packages under this folder will be available for use to any Go projects under "github.com/foo/*"